test/SimpleSaslTls.cpp

```diff
@@ -0,0 +1,173 @@
+#include "infinispan/hotrod/ConfigurationBuilder.h"
+#include "infinispan/hotrod/RemoteCacheManager.h"
+#include "infinispan/hotrod/RemoteCache.h"
+#include "infinispan/hotrod/Version.h"
+
+#include "infinispan/hotrod/JBasicMarshaller.h"
+#include <sasl/saslplug.h>
+
+#include <stdlib.h>
+#include <iostream>
+#include <memory>
+#include <typeinfo>
+#include <thread>
+#include <future>
+#include <chrono>
+
+// For CTest: return 0 if all tests pass, non-zero otherwise.
+
+using namespace infinispan::hotrod;
+
+namespace infinispan {
+namespace hotrod {
+namespace transport {
+
+class MyRoundRobinBalancingStrategy: public FailOverRequestBalancingStrategy {
+public:
+    MyRoundRobinBalancingStrategy() :
+            index(0) {
+    }
+
+    static FailOverRequestBalancingStrategy *newInstance() {
+        return new MyRoundRobinBalancingStrategy();
+    }
+
+    void setServers(const std::vector<transport::InetSocketAddress> &s) {
+        servers = s;
+        // keep the old index if possible so that we don't produce more requests for the first server
+        if (index >= servers.size()) {
+            index = 0;
+        }
+    }
+
+    ~MyRoundRobinBalancingStrategy() {
+    }
+
+    const transport::InetSocketAddress &getServerByIndex(size_t pos) {
+        const transport::InetSocketAddress &server = servers[pos];
+        return server;
+    }
+private:
+    std::vector<transport::InetSocketAddress> servers;
+    size_t index;
+    const transport::InetSocketAddress &nextServer(const std::set<transport::InetSocketAddress>& failedServers) {
+        for (unsigned int i = 0; i <= servers.size(); i++) {
+            const transport::InetSocketAddress &server = getServerByIndex(index++);
+            if (failedServers.empty() || failedServers.find(server) != failedServers.end()
+                    || i > failedServers.size()) {
+                if (index >= servers.size()) {
+                    index = 0;
+                }
+            }
+            return server;
+        }
+
+        throw Exception("Bad news, no server available.");
+    }
+};
+
+}
+}
+}
+
+template<class T>
+void assert_not_null(const std::string& message, int line, const std::unique_ptr<T>& pointer) {
+    if (pointer.get() == 0) {
+        std::cerr << message << ":" << line << std::endl;
+        std::cerr.flush();
+        throw std::exception();
+    }
+}
+
+static char *simple_data; // plain
+static char realm_data[] = "applicationRealm";
+static char path_data[] = "/usr/lib64/sasl2";
```

Comment:
The #define below is the same. Also on Debian/Ubuntu this is different (/usr/lib/x86_64-linux-gnu/sasl2/). The default build will scan the known paths already without the need to supply them and it can be overridden with the SASL_PATH env var, so maybe we shouldn't touch this at all to increase portability